File: An 2/Sem 1/MAP/Lab6/C#/Lab2/Model/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2
{
	namespace Lab2_Model {
	    class Student : ComparableStudent
	    {
	        public string name;
	        public int id;
	        public int grade;

	        public Student(int id, string name, int grade)
	        {
	            this.id = id;
	            this.name = name;
	            this.grade = grade;
	        }

			public virtual float average() {
				return grade;
			}

			bool ComparableStudent.isGreaterThan(Student student) {
				return (this.grade > student.grade);
			}
	    }
	}
}

File: An 2/Sem 1/MAP/Lab6/Java/src/lab2/Model/Student.java
package lab2.Model;

/**
 *
 * @author mihai
 */
public class Student implements Comparable<Student> {
    public String name;
    public int id;
    public int grade;
    
    public Student(int id, String name, int grade){
        this.name = name;
        this.id = id;
        this.grade = grade;
    }

    public float average() {
        return grade;
    }

    @Override
    public boolean isGreaterThan(Student student) {
        return (this.grade > student.grade);
    }

    @Override
    public String toString() {
        return String.format("%d|%s|%d|", this.id, this.name, this.grade);
    }
}
